perfkitbenchmarker/configs/option_decoders.py

```diff
@@ -83,6 +83,55 @@ def Decode(self, value, component_full_name, flag_values):
     raise NotImplementedError()
 
 
+class EnumDecoder(ConfigOptionDecoder):
+  """ Verifies that the config options value is on of the allowed set.
+
+  Passes through the value unmodified
+  """
+
+  def __init__(self, option, none_ok=False, valid_values=[], **kwargs):
```

Comment:
It's dangerous to use [] for the default value of a list. Python will create it once when the function is defined and re-use the same list instance for every call of the function when a value is not explicitly provided. In general, your intent is to create a new list each time the function is called, and the way to do that is by setting the default to None and initializing like so:
```
self.valid_values = valid_values or []
```
In this specific case, it may not make sense to have a default, considering that would mean the decoder doesn't accept any values.